DSharpPlus.Commands/Processors/SlashCommands/SlashCommandProcessor.cs

```diff
@@ -0,0 +1,442 @@
+
+        Dictionary<ulong, Command> commandsDictionary = [];
+        foreach (DiscordApplicationCommand discordCommand in discordCommands)
+        {
+            if (!extension.Commands.TryGetValue(discordCommand.Name, out Command? command))
+            {
+                foreach (Command aliasedCommand in extension.Commands.Values)
+                {
+                    if (aliasedCommand.Attributes.OfType<DisplayNameAttribute>().FirstOrDefault()?.DisplayName == discordCommand.Name)
+                    {
+                        command = aliasedCommand;
+                        break;
+                    }
+                }
+            }
+
+            if (command is null)
+            {
+                SlashLogging.UnknownCommandName(this._logger, discordCommand.Name, null);
+                continue;
+            }
+
+            commandsDictionary.Add(discordCommand.Id, command);
+        }
+
+        this.Commands = commandsDictionary.ToFrozenDictionary();
+        SlashLogging.RegisteredCommands(this._logger, this.Commands.Count, null);
+    }
+
+    [SuppressMessage("Roslyn", "CA1859", Justification = "Incorrect warning in NET 8-rc2. TODO: Open an issue in dotnet/runtime.")]
+    public async Task<DiscordApplicationCommand> ToApplicationCommandAsync(Command command)
+    {
+        if (this._extension is null)
+        {
+            throw new InvalidOperationException("SlashCommandProcessor has not been configured.");
+        }
+
+        // Translate the command's name and description.
+        IReadOnlyDictionary<string, string> nameLocalizations = new Dictionary<string, string>();
+        IReadOnlyDictionary<string, string> descriptionLocalizations = new Dictionary<string, string>();
+        if (command.Attributes.OfType<SlashLocalizerAttribute>().FirstOrDefault() is SlashLocalizerAttribute localizerAttribute)
+        {
+            AsyncServiceScope scope = this._extension.ServiceProvider.CreateAsyncScope();
+            nameLocalizations = await localizerAttribute.LocalizeAsync(scope.ServiceProvider, $"{command.FullName}.name");
+            descriptionLocalizations = await localizerAttribute.LocalizeAsync(scope.ServiceProvider, $"{command.FullName}.description");
+            await scope.DisposeAsync();
+        }
+
+        // Convert the subcommands or parameters into application options
+        List<DiscordApplicationCommandOption> options = [];
+        if (command.Subcommands.Any())
+        {
+            foreach (Command subCommand in command.Subcommands)
+            {
+                options.Add(await this.ToApplicationParameterAsync(subCommand));
+            }
+        }
+        else
+        {
+            foreach (CommandParameter parameter in command.Parameters)
+            {
+                if (parameter.Attributes.OfType<ParamArrayAttribute>().Any())
+                {
+                    // Fill til 25
```

Comment:
do we not want to support an upper bound?

Reply:
We do. I was half expecting to rewrite this specific part of the code later through checks.